packages/typescript/index.bzl

```diff
@@ -63,6 +63,7 @@ def ts_project(
         out_dir = None,
         root_dir = None,
         link_workspace_root = False,
+        module_type = "commonjs",
```

Comment:
could you separate the TS api changes from the rest of the PR? This part worries me since it adds complexity and bazel-only behaviors, in other ways this rule "just runs `tsc`". But I think the other parts could land.

Reply:
Yeah, really not sure what the solution is here, happy to remove and at least land the other pieces first